src/Subscriptions/Models/Subscription.php

```diff
@@ -172,6 +172,36 @@ public static function create(array $attributes): Subscription
         return $subscription;
     }
 
+    /**
+     * @unreleased
+     * @throws Exception
+     */
+    public function createRenewal(array $attributes = []): Donation
+    {
+        return give()->subscriptions->createRenewal($this, $attributes);
+    }
+
+    /**
+     * @unreleased
+     */
+    public function shouldCreateRenewal(): bool
+    {
+        $billTimes = $this->installments;
+        $totalPayments = count($this->donations);
```

Comment:
This PR looks handy, @jonwaldstein! What do you think about making a `$subscription->totalDonations()` method to make this declarative (and in the other method)? Also, does `$this->donations` query and instantiate every single donation? That could be quite costly for some subscriptions (e.g. giving weekly for years). Is it possible to do something like `$this->donations()->count()`?

Reply:
Oh yeah, good call i'll update it

Reply:
updated c719b1bd